### PR TITLE
Fix lint job crash when no packages were changed

### LINKED ISSUE #361: Avoid running the lint step if no packages were changed
This [PR](https://github.com/ocaml/opam-repository/pull/26508), for instance, doesn't change any packages and the analysis correctly detects it. But, we still try to run a lint with no packages, and [fail](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/5850dbfbc9a7b287c7242765841029ed8469fc30/variant/(lint))

### PR DESCRIPTION
**Pull request by punchagan**
Closes #361

### CODE REVIEW

File: lib/lint.ml [hunk 1]
```diff
@@ -77,8 +77,13 @@ module Lint = struct
...
     Current.Job.start job ~pool ~level:Current.Level.Harmless >>= fun () ->
-    Current_git.with_checkout ~job src @@ fun dir ->
-    Check.of_dir ~master ~job ~packages dir
+    match List.length packages with
+    | 0 ->
+      Current.Job.log job "No packages to lint.";
+      Lwt.return (Ok ())
+    | _ ->
+      Current_git.with_checkout ~job src @@ fun dir ->
+      Check.of_dir ~master ~job ~packages dir
```
Comment (shonfeder) [inline lib/lint.ml:86]: `List.length` needs to count all the elements of a list. Emptiness are checks are therefore better done by pattern matching, which take a single operation:

```suggestion
    match packages with
    | [] ->
      Current.Job.log job "No packages to lint.";
      Lwt.return (Ok ())
    | _ ->
      Current_git.with_checkout ~job src @@ fun dir ->
      Check.of_dir ~master ~job ~packages dir
```

This is also how [`List.is_empty`](https://github.com/ocaml/ocaml/blob/d753ea1992d27aa22dcb449f4a0ac178d167d52c/stdlib/list.ml#L571-L573) works.

**Review by shonfeder — APPROVED** [review]
The general approach here seems fine to me! I think is solution is nice, because we may end up wanting to run lint checks on thinks other than opam files.

**Review by shonfeder — CHANGES_REQUESTED** [review]
Just the one list checking idiom request on my end!

File: lib/lint.ml (hunk 1 above)
Comment (punchagan) [reply lib/lint.ml:86]: Thanks for the mini-tutorial! Appreciate it. :)

**Review by shonfeder — APPROVED** [review]

**Pull request merged**